File: DSA_CSHARP/Matrices/TridiagonalMatrix.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace DSA_CSHARP.Matrices;
internal class TridiagonalMatrix
{
    //A tridiagonal matrix is a special type of square matrix in which only the elements on the main diagonal, the diagonal just above it, and the diagonal just below it can be non-zero.
    //All other elements must be zero.
    static void ReadMatrix(int[,] matrix)
    {
        int n = matrix.GetLength(0);
        Console.WriteLine("Enter matrix elements:");
        Console.WriteLine("Note: Only main, upper, and lower diagonals are allowed to be non-zero.");
        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j < n; j++)
            {
                if (Math.Abs(i - j) <= 1) // i==j, i==j-1, i==j+1
                {
                    Console.Write($"Element [{i},{j}]: ");
                    matrix[i, j] = int.Parse(Console.ReadLine());
                }
                else
                {
                    matrix[i, j] = 0; 
                }
            }
        }
    }

    static void PrintMatrix(int[,] matrix)
    {
        int n = matrix.GetLength(0);
        Console.WriteLine("\nTridiagonal Matrix Output:");
        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j < n; j++)
            {
                Console.Write(matrix[i, j] + " ");
            }
            Console.WriteLine();
        }
    }

    public string Main()
    {
        Console.Write("Enter the size of the square matrix (n x n): ");
        int n = int.Parse(Console.ReadLine());

        int[,] matrix = new int[n, n];
        ReadMatrix(matrix);
        PrintMatrix(matrix);
        return "";
    }
}

File: DSA_PYTHON/Matrices/TridiagonalMatrix.py

class TridiagonalMatrix:
    # A tridiagonal matrix is a special type of square matrix in which only the elements on the main diagonal, the diagonal just above it, and the diagonal just below it can be non-zero.
    # All other elements must be zero.
    def read_matrix(self, n):
        matrix = [[0 for _ in range(n)] for _ in range(n)]
        print("Enter elements for Tridiagonal Matrix:")
        print("Note: Only fill positions where |i - j| <= 1")

        for i in range(n):
            for j in range(n):
                if abs(i - j) <= 1:
                    val = int(input(f"Element [{i}][{j}]: "))
                    matrix[i][j] = val
                else:
                    matrix[i][j] = 0
        return matrix

    def print_matrix(self, matrix):
        print("\nTridiagonal Matrix Output:")
        for row in matrix:
            for val in row:
                print(val, end=" ")
            print()

    def main(self):
        n = int(input("Enter the size of the square matrix (n x n): "))
        matrix = self.read_matrix(n)
        self.print_matrix(matrix)
